test/functional/tests.py

```diff
@@ -456,7 +456,7 @@ def setUp(self):
         self._olddir = os.getcwd()
         os.chdir(HERE)
         # Add a resource to get the stats for.
-        self.resource = self.api.add_str(u'Mary had a little lamb')
```

Comment:
The `add_str` method still allows binary strings on Python 2 (but not on Python 3), because of backwards compatibility concerns and because it's, in my experience, very normal to use UTF-8 encoded binary strings in Python 2 instead of proper `unicode`.
By having the string be binary on Python 2 and text on Python 3, it's testing that the *de facto* preferred string type on each platform works properly. It's, in a way, the same reason why there used to be no `add_bytes` method at all.